boot/seal.go

```diff
@@ -261,14 +261,28 @@ func sealKeyToModeenvForMethod(
 
 var resealKeyToModeenv = resealKeyToModeenvImpl
 
+// ResealKeyToModeenvOptions are options to pass to resealing which is
+// not related to modeenv or boot chains.
+type ResealKeyToModeenvOptions struct {
+	// ExpectReseal must be passed when calling boot.IsResealNeeded
```

Comment:
ExpectReseal is set true when a reseal is usually expected, it is be used when consulting boot.IsResealNeeded which uses it to disambiguate cases where it cannot be fully determined if measurements have changed